CONTRIBUTING.md

```diff
@@ -5,13 +5,55 @@ email, or any other method with the owners of this repository before making a ch
 
 Please note we have a [code of conduct](CODE_OF_CONDUCT.md) that we ask you to follow in all your interactions with the project.
 
+**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
+
+*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*
+
+Thank you for your pull request. Please provide a description above and review
+the requirements below.
+
 ## Pull Request Process
 
+0. Check out [Pull Request Checklist](#pull-request-checklist), ensure you have fulfilled each step.
 1. Ensure any install or build dependencies are removed before the end of the layer when doing a
    build.
-2. Update the README.md with details of changes to the interface, this includes new environment
+2. Please ensure the [README](README.md) and [DOCS](./DOCS.md) are up-to-date with details of changes to the interface, this includes new environment.
    variables, exposed ports, useful file locations and container parameters.
-3. Increase the version numbers in any examples files and the README.md to the new version that this
-   Pull Request would represent. The versioning scheme we use is [SemVer](http://semver.org/).
-4. You may merge the Pull Request in once you have the sign-off of two other developers, or if you
-   do not have permission to do that, you may request the second reviewer to merge it for you.
+3. **PLEASE ENSURE YOU DO NOT INTRODUCE BREAKING CHANGES.**
+4. **PLEASE ENSURE BUG FIXES AND NEW FEATURES INCLUDE TESTS.**
+5. You may merge the Pull Request in once you have the sign-off of one other maintainer/code owner,
+   or if you do not have permission to do that, you may request the second reviewer to merge it for you.
+
+## Pull Request Checklist
+
+- [x] Read the **CONTRIBUTING** document. (It's checked since you are alread here.)
+- [ ] Read the **CODE OF CONDUCT** document.
+- [ ] Add tests to cover changes.
+- [ ] Make sure your code follows the code style of this project
+- [ ] Make sure CI and all other PR checks are green OR
+    - [ ] Code compiles correctly.
+    - [ ] Created tests which fail without the change (if possible).
+    - [ ] All new and existing tests passed.
+- [ ] Add your changes to [CHANGELOG](CHANGELOG.md).
+- [ ] Improve and update the [README](README.md) (if necessary).
+- [ ] Make sure [documentation](./DOCS.md) is up-to-date. Same file should also be updated in [plugin index](https://github.com/drone/drone-plugin-index/blob/master/content/meltwater/drone-cache/index.md) by sending a PR to that repository.
```

Comment:
When a contributor sends a PR to `drone-cache` they don't actually know if their PR will be merged or when it will be merged. So in short they don't know if their changes will go live.

Therefore I think that a contributor cannot sensibly update https://github.com/drone/drone-plugin-index, can they?

Reply:
No, you are definetly right. That part should go to `release` section and it should be maintainers' responsibility.

It should be rephrased:
```suggestion
- [ ] Make sure [documentation](./DOCS.md) is up-to-date. The same file will be updated in [plugin index](https://github.com/drone/drone-plugin-index/blob/master/content/meltwater/drone-cache/index.md) when your PR is accepted, so it will be available for end-users at http://plugins.drone.io
```